src/chart/generateCategoricalChart.tsx

```diff
@@ -1322,21 +1298,23 @@ export const generateCategoricalChart = ({
 
     /**
      * Get the information of mouse in chart, return null when the mouse is not in the chart
-     * @param  {Object} event    The event object
+     * @param  {MousePointer} event    The event object
      * @return {Object}          Mouse data
      */
-    getMouseInfo(event: React.MouseEvent | React.Touch) {
+    getMouseInfo(event: MousePointer) {
       if (!this.container) {
         return null;
       }
 
       const containerOffset = getOffset(this.container);
-      const e = calculateChartCoordinate(event, containerOffset);
+      const e = {
+        chartX: Math.round(event.pageX - containerOffset.left),
+        chartY: Math.round(event.pageY - containerOffset.top),
+      };
```

Comment:
why remove this? These are the same and the function is much easier to test in isolation than this

Reply:
Personally, I don't think this function needs to be tested, and at least for now, as it is not reused, I think it's not a bad idea to do this for readability

Reply:
Hm yeah this case is fine I guess - don't want to do too many things inline or without tests else we'll end up worse than we are now with generateCategoricalChart, but this instance should be fine